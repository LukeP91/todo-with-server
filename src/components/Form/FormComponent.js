import { Formik } from 'formik'
import PropTypes from 'prop-types'
import React, { Component } from 'react'

import schema from './schema'
import TodoForm from './TodoForm'

export default class FormComponent extends Component {
  render() {
    const { users, labels, initialValues, submit } = this.props

    return (
      <Formik
        initialValues={initialValues}
        onSubmit={submit}
        render={formikProps => <TodoForm {...formikProps} labels={labels} users={users} />}
        validationSchema={schema}
      />
    )
  }
}

FormComponent.propTypes = {
  initialValues: PropTypes.object,
  labels: PropTypes.array,
  submit: PropTypes.func,
  users: PropTypes.array,
}

