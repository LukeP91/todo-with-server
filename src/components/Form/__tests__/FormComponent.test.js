import React from 'react'
import renderer from 'react-test-renderer'

import FormComponent from '../FormComponent'

describe('FormComponent', () => {
  it('renders correctly', () => {
    const props = {
      users: [{ id: 1, name: 'Joe Doe' }],
      labels: [{ id: 1, value: 'test' }],
      initialValues: { title: '', description: '', userId: '', labelsIds: [] },
      submit: jest.fn(),
    }
    const wrapper = renderer.create(<FormComponent {...props} />).toJSON()

    expect(wrapper).toMatchSnapshot()
  })
})
