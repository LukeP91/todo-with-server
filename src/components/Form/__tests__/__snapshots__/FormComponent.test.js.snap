// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormComponent renders correctly 1`] = `
<form
  className="TodoForm-container-1"
  onSubmit={[Function]}
>
  <div
    className="MuiFormControl-root-3 TextInput-container-2"
  >
    <label
      className="MuiFormLabel-root-12 MuiInputLabel-root-7 MuiInputLabel-formControl-8 MuiInputLabel-animated-11"
      data-shrink={false}
      htmlFor="name-simple"
    >
      Title
    </label>
    <div
      className="MuiInput-root-19 MuiInput-formControl-20 MuiInput-underline-23"
    >
      <input
        aria-invalid={false}
        className="MuiInput-input-27"
        disabled={false}
        id="name-simple"
        name="title"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    className="MuiFormControl-root-3 TextInput-container-2"
  >
    <label
      className="MuiFormLabel-root-12 MuiInputLabel-root-7 MuiInputLabel-formControl-8 MuiInputLabel-animated-11"
      data-shrink={false}
      htmlFor="name-simple"
    >
      Description
    </label>
    <div
      className="MuiInput-root-19 MuiInput-formControl-20 MuiInput-underline-23"
    >
      <input
        aria-invalid={false}
        className="MuiInput-input-27"
        disabled={false}
        id="name-simple"
        name="description"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    className="MuiFormControl-root-3 TextInput-container-32"
  >
    <label
      className="MuiFormLabel-root-12 MuiInputLabel-root-7 MuiInputLabel-formControl-8 MuiInputLabel-animated-11"
      data-shrink={false}
      htmlFor="name-simple"
    >
      User
    </label>
    <div
      className="MuiInput-root-19 MuiInput-formControl-20 MuiInput-underline-23"
    >
      <div
        className="MuiSelect-root-33"
      >
        <div
          aria-haspopup="true"
          aria-owns={null}
          aria-pressed="false"
          className="MuiSelect-select-34 MuiSelect-selectMenu-35 MuiInput-input-27"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={0}
        >
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "&#8203;",
              }
            }
          />
        </div>
        <input
          id="name-error"
          name="userId"
          type="hidden"
          value=""
        />
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-38 MuiSelect-icon-37"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </div>
    </div>
  </div>
  <div
    className="MultipleSelect-root-49"
  >
    <div
      className="MuiFormControl-root-3 MultipleSelect-formControl-50"
    >
      <label
        className="MuiFormLabel-root-12 MuiInputLabel-root-7 MuiInputLabel-formControl-8 MuiInputLabel-animated-11"
        data-shrink={false}
        htmlFor="select-multiple-chip"
      >
        Labels
      </label>
      <div
        className="MuiInput-root-19 MuiInput-formControl-20 MuiInput-underline-23"
      >
        <div
          className="MuiSelect-root-33"
        >
          <div
            aria-haspopup="true"
            aria-owns={null}
            aria-pressed="false"
            className="MuiSelect-select-34 MuiSelect-selectMenu-35 MuiInput-input-27"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <span
              dangerouslySetInnerHTML={
                Object {
                  "__html": "&#8203;",
                }
              }
            />
          </div>
          <input
            id="labelsIds"
            name="labelsIds"
            type="hidden"
            value=""
          />
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-38 MuiSelect-icon-37"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
  <button
    className="MuiButtonBase-root-79 MuiButton-root-53 MuiButton-contained-64 MuiButton-containedPrimary-65 MuiButton-raised-67 MuiButton-raisedPrimary-68"
    disabled={false}
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="submit"
  >
    <span
      className="MuiButton-label-54"
    >
      Submit
    </span>
    <span
      className="MuiTouchRipple-root-82"
    />
  </button>
</form>
`;
